packages/flutter/lib/src/gestures/tap.dart

```diff
@@ -98,6 +98,197 @@ typedef GestureTapCallback = void Function();
 ///  * [TapGestureRecognizer], which uses this signature in one of its callbacks.
 typedef GestureTapCancelCallback = void Function();
 
+/// The base class of gesture recognizers that recognize taps.
+///
+/// Gesture recognizers take part in gesture arenas to enable potential gestures
+/// to be disambiguated from each other. This process is managed by a
+/// [GestureArenaManager].
+///
+/// A tap is defined as a sequence of events that starts with a down, followed
+/// by optional moves, then ends with an up. All move events must contain the
+/// same `buttons` with the down event, and must not be too far from the initial
```

Comment:
```suggestion
/// same `buttons` as the down event, and must not be too far from the initial
```

Reply:
Buttons for the up event don't matter? Do all buttons have to come up at the same time?

Reply:
1. Up events always have `buttons: 0`.
2. If the pointer's `buttons` changes to a different non-zero value, it's considered a Move event. (for example, if `buttons` changes from `2 | 1` to `1`, then there will be a Move event with `buttons: 1`)

Therefore, since multiple buttons will almost always be pressed or released one at a time, `BaseTapGestureRecognizer` will not accept multi-button taps (which was intentionally designed when we added button constraint to gestures)